### DISCUSSION TITLE
The first fuzzy query after Fuseki started is particularly slow

### DISCUSSION THREAD

**Discussion by chenkuncloud** [apache/jena#1418] [Q&A] [answered]
Hi,

The first fuzzy query after Fuseki started is particularly slow, taking about 17 seconds.

But the next query (the same SPARQL) only takes 2 seconds.

I changed 'w' to 'wa' or 'water', queries are still faster.

And I've found that happens over time.

What's the reason? Can it be preloaded to speed up each fuzzy query? 

Here's my SPARQL:
```
PREFIX ...

SELECT ( COUNT( DISTINCT ?s ) AS ?count)
WHERE {
	GRAPH ?graph {
		{
			?s ontology:field_A ?field_A .
			FILTER ( CONTAINS(?field_A, 'w') ) .
		}
		UNION
		{
			?s ontology:field_B ?field_B .
			FILTER ( CONTAINS(?field_B, 'w') ) .
		}
		UNION
		{
			?s ontology:field_C ?field_C .
			FILTER ( CONTAINS(?field_C, 'w') ) .
		}
	}
}
```

Here's some information that may help :
- Environment: CentOS 7 + fuseki-4.4.0 + openjdk 18
- RDF Triples: 4 million
- count: 5000
      
Best Regards

**Comment by afs** [comment, marked as answer]
There are two effects here.

Caches of on-disk data start empty and fill up on demand when using TDB1 or TDB2. This is both in the operating system and and node table cache. The effect is less pronounced when data is stored on a local SSD. The more the access latency the slower the first usage will be.

The JDK will be compiling and optimizing bytecode on-the-fly (the JIT compiler).

To preload, send some queries to the server and startup.

**Comment by chenkuncloud** [comment]
Thanks.

After listening to your reply, I adopted the plan: Send fuzzy query to fuseki at set intervals.
